### PR TITLE
Migrate to using common-files and build tools image for linting

### PR DESCRIPTION
**Pull request by ericvn**
This PR moves the istio/community repo to using the same `common-files` as the rest of the Istio repositories. This removes the 3 year old build tools image referenced in the prow/community-lint.sh.

This change shouldn't require much more from someone trying to run lint (already required `docker`) before submitting a PR. They would need `make`, but wouldn't need `Go`. I don't think this is a showstopper. 

I did make the updates to pass the newer linting in the current build tools image.

Follow-up changes in other repos:
 - test-infra:
     - change the lint job to use make instead of calling the script directly.
     - change the `update-common-mainonly` job to include the community repo.
 - common-files: 
    - update the update-common target to not copy CONTRIBUTING.MD if the current version doesn't have the referral to istio/community.
    - Update the LICENSE file copyright

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,191 +1,5 @@
...
-Are you running Istio in production?  List yourself on our [Case Studies](https://istio.io/latest/about/case-studies/) page.
-
-If you want to add your logo to our wall of users, add an entry to the `users`
-section in [data/companies.yml](https://github.com/istio/istio.io/blob/master/data/companies.yml), and add your logo,
-preferably in SVG format, to [static/logos](https://github.com/istio/istio.io/blob/master/static/logos).
-
-If you'd like to be interviewed for a full case study, please [create an issue](https://github.com/istio/istio.io/issues/new),
-and we'll be in touch!
+So you want to hack on Istio? Yay! Please refer to Istio's overall
+[contribution guidelines](https://github.com/istio/community/blob/master/CONTRIBUTING.md)
```
Comment (ericvn) [inline CONTRIBUTING.md:4]: This is the link that needs updating in common-files (point to renamed file):
```suggestion
[contribution guidelines](https://github.com/istio/community/blob/master/CONTRIBUTION-GUIDELINES.md)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,191 +1,5 @@
...
+to find out how you can help.
```
Comment (craigbox) [inline CONTRIBUTING.md:5]: This change makes this file self-referential?

I would like the content to remain in a file named CONTRIBUTING.md if possible; that's the standard that most projects use.

File: LICENSE [hunk 3]
```diff
@@ -186,7 +187,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright {yyyy} {name of copyright owner}
+   Copyright 2016-2020 Istio Authors
```
Comment (craigbox) [inline LICENSE:190]: 2022

**Review by craigbox — COMMENTED** [review]
I don't fully understand the implications of this change, but I'm surprised to see all the licenses in the community repo.

File: LICENSE (hunk 3 above)
Comment (ericvn) [reply LICENSE:190]: This is from https://github.com/istio/common-files/blob/master/files/LICENSE#L190. Would need to update it there. Again, this file gets copied to each repo as part of the `make update-common`.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ericvn) [reply CONTRIBUTING.md]: Looking at the makefile, there is no way to determine which repo is doing the `make update-common`. Thus, I can't stop the copying of the file.

I could remove the file from common-files altogether which would prevent overriding the CONTRIBUTION.md file in the repos and everything stays the same. The negative here is if we ever wanted to update the CONTRIBUTION.md file in the other repos we would have to do it manually (which isn't that hard, just ~10 PRs, but I don't expect to happen all that often, if at all.)

**Comment by ericvn** [comment]
> I don't fully understand the implications of this change, but I'm surprised to see all the licenses in the community repo.

I removed the license directory

File: CONTRIBUTING.md (hunk 2 above)
Comment (ericvn) [reply CONTRIBUTING.md]: Here's my current change (which will get integrated into the common-files) changing to only copy from common-files if the current file has a `refer to Istio` in it (this is part of the current common CONTRIBUTING.md). This allows the CONTRIBUTING.md in istio/community to remain the long version, while the other repos have a shorter version pointing to the istio/community file.

In the future, if we change the common-files CONTRIBUTING.md, we may need to tweak things depending on the text there.

**Comment by ericvn** [comment]
@craigbox I think I have the issues resolved now. If you look at https://github.com/ericvn/community/tree/convertToBuildContainer the CONTRIBUTING.md continues to be the full version.

The other changes are the addition of the common-files repo files related to building, and adding other files like SUPPORT.md.

A follow-on PR to common-files is needed with my change to not override the CONTRIBUTING.md as noted in my prior note, plus updating the LICENSE file per your earlier remark.

**Review by craigbox — APPROVED** [review]

**Pull request merged**